transports/tls/tls.go

```diff
@@ -2,27 +2,120 @@ package tls
 
 import (
 	"crypto/tls"
+	"crypto/x509"
+	"io/ioutil"
+	"log"
 	"net"
+	"os"
+	"path/filepath"
 
 	"github.com/gliderlabs/logspout/adapters/raw"
 	"github.com/gliderlabs/logspout/router"
 )
 
+// default CA and certificate path
+const (
+	CAPath   = "/mnt/ca/"
+	CertPath = "/mnt/cert/"
+)
+
 func init() {
 	router.AdapterTransports.Register(new(tlsTransport), "tls")
 	// convenience adapters around raw adapter
 	router.AdapterFactories.Register(rawTLSAdapter, "tls")
 }
 
+func getopt(name, dfault string) string {
+	value := os.Getenv(name)
+	if value == "" {
+		value = dfault
+	}
+	return value
+}
+
 func rawTLSAdapter(route *router.Route) (router.LogAdapter, error) {
 	route.Adapter = "raw+tls"
 	return raw.NewRawAdapter(route)
 }
 
+func getCAs(path string) *x509.CertPool {
+	cafiles, err := ioutil.ReadDir(path)
+	if err != nil {
+		return nil
+	}
+	capool := x509.NewCertPool()
+	// TODO: don't khow how to check pool is empty
+	empty := true
```

Comment:
empty can be gotten rid of by checking `len(capool.Subjects()) != 0`, see https://play.golang.org/p/RrfPNW9fPh